ribbon-core/src/main/java/com/netflix/client/PrimeConnections.java

```diff
@@ -286,8 +303,12 @@ public Void call() throws Exception {
                 try {
                     ftC = makeConnectionASync(s, listener);
                     ftList.add(ftC);
-
-                } catch (Throwable e) { // NOPMD
+                }
+                catch (RejectedExecutionException ree) {
+                    logger.error("executor submit failed", ree);
+                }
+                catch (Throwable e) {
```

Comment:
@elandau agreed.  in talking with @allenxwang we wanted to be careful of changing the behavior here (orig code only caught Throwable), but if you think it's safe, I'm for it as well.  The fact that it was catching Throwable and the async code change I believe caused the bug I was fixing.